weaviate/collections/batch/base.py

```diff
@@ -57,16 +67,18 @@ def __init__(self) -> None:
     def __len__(self) -> int:
         return len(self.__items)
 
-    def clear(self) -> None:
-        """Remove all the items from the BatchRequest."""
-        self.__items.clear()
-
     def add(self, item: TBatchInput) -> None:
         """Add an item to the BatchRequest."""
         self.__lock.acquire()
         self.__items.append(item)
         self.__lock.release()
 
+    def prepend(self, item: List[TBatchInput]) -> None:
+        """Add items to the front of the BatchRequest."""
+        self.__lock.acquire()
+        self.__items = item + self.__items
+        self.__lock.release()
```

Comment:
If this is a requirement of the algo, I think `self.__items` should be converted to a `deque` so that prepends are `O(1)` rather than `O(n)`. WDYT?

Reply:
Okay, makes sense! Looking at how to slice a deque, it's a lot more code and deps so no problem here :D